citadel_workspace_service/src/kernel/mod.rs

```diff
@@ -156,58 +183,69 @@ async fn payload_handler(
                 Ok(_res) => {
                     // TODO: add trace ID to ensure uniqueness of request
                     let response = InternalServiceResponse::RegisterSuccess { id: uuid };
-                    send_response_to_tcp_client(hm, response, uuid).await
+                    send_response_to_tcp_client(tcp_connection_map, response, uuid).await
                 }
                 Err(err) => {
                     let response = InternalServiceResponse::RegisterFailure {
                         message: err.to_string(),
                     };
-                    send_response_to_tcp_client(hm, response, uuid).await
+                    send_response_to_tcp_client(tcp_connection_map, response, uuid).await
                 }
             };
         }
         InternalServicePayload::Message {
+            uuid: _,
             message,
             cid,
-            user_cid: _,
+            user_cid,
             security_level,
         } => {
-            match connection_map.get_mut(&cid) {
-                Some(sink) => {
-                    sink.set_security_level(security_level);
-                    sink.send_message(message.into()).await.unwrap();
+            match server_connection_map.get_mut(&cid) {
+                Some(remote) => {
+                    remote.sink.set_security_level(security_level);
+
+                    let _response = InternalServiceResponse::MessageReceived {
```

Comment:
We need to tell the TCP client that the message request has been received by the internal service.
```suggestion
                    let _response = InternalServiceResponse::MessageSent {
```